File: basic recursion/javascript/n_times_name.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function repeatName(name, n) {
    if (n <= 0) {
        return "";
    }
    return name + repeatName(name, n - 1);
}

rl.question("Enter the number of times to repeat: ", function(n) {
    n = parseInt(n);
    name = "name_repeat "
    if (isNaN(n) || n <= 0) {
        console.log("Please enter a valid positive integer.");
    } else {
        const result = repeatName(name, n);
        console.log(`The name "${name}" repeated ${n} times is:`)
        console.log(result);
    }
    rl.close();
});

File: basic recursion/c++/n_times_name.cpp
#include <iostream>
using namespace std;

void printName(int n, string name)
{
    if (n == 0)
    {
        return;
    }
    cout << name << " ";
    printName(n - 1, name);
}

int main()
{
    int n;
    string name = "to_repeat";
    cout << "Enter a number: ";
    cin >> n;
    printName(n, name);
    return 0;
}